Xamarin.Forms.Platform.iOS/Material/MaterialStepperRenderer.cs

```diff
@@ -0,0 +1,154 @@
+using Xamarin.Forms;
+using MButton = MaterialComponents.Button;
+
+[assembly: ExportRenderer(typeof(Xamarin.Forms.Stepper), typeof(Xamarin.Forms.Platform.iOS.Material.MaterialStepperRenderer), new[] { typeof(VisualRendererMarker.Material) })]
+
+namespace Xamarin.Forms.Platform.iOS.Material
+{
+	public class MaterialStepperRenderer : ViewRenderer<Stepper, MaterialStepper>
+	{
+		ButtonScheme _buttonScheme;
+
+		public MaterialStepperRenderer()
+		{
+			VisualElement.VerifyVisualFlagEnabled();
+		}
+
+		protected override void Dispose(bool disposing)
+		{
+			if (Control is MaterialStepper control)
+			{
+				control.DecrementButton.TouchUpInside -= OnStep;
+				control.IncrementButton.TouchUpInside -= OnStep;
+			}
+
+			base.Dispose(disposing);
+		}
+
+		protected override void OnElementChanged(ElementChangedEventArgs<Stepper> e)
+		{
+			_buttonScheme?.Dispose();
+			_buttonScheme = CreateButtonScheme();
+
+			base.OnElementChanged(e);
+
+			if (e.NewElement != null)
+			{
+				if (Control == null)
+				{
+					var stepper = CreateNativeControl();
+					stepper.DecrementButton.TouchUpInside += OnStep;
+					stepper.IncrementButton.TouchUpInside += OnStep;
+					SetNativeControl(stepper);
+				}
+
+				UpdateButtons();
+				ApplyTheme();
+			}
+		}
+
+		protected virtual ButtonScheme CreateButtonScheme()
+		{
+			return new ButtonScheme
+			{
+				ColorScheme = MaterialColors.Light.CreateColorScheme(),
+				ShapeScheme = new ShapeScheme(),
+				TypographyScheme = new TypographyScheme(),
+			};
+		}
+
+		protected virtual void ApplyTheme()
+		{
+			OutlinedButtonThemer.ApplyScheme(_buttonScheme, Control.DecrementButton);
+			OutlinedButtonThemer.ApplyScheme(_buttonScheme, Control.IncrementButton);
+		}
+
+		protected override MaterialStepper CreateNativeControl()
+		{
+			return new MaterialStepper();
+		}
+
+		protected override void OnElementPropertyChanged(object sender, PropertyChangedEventArgs e)
+		{
+			base.OnElementPropertyChanged(sender, e);
+
+			if (e.PropertyName == Stepper.MinimumProperty.PropertyName ||
+				e.PropertyName == Stepper.MaximumProperty.PropertyName ||
+				e.PropertyName == Stepper.ValueProperty.PropertyName ||
+				e.PropertyName == VisualElement.IsEnabledProperty.PropertyName)
+			{
+				UpdateButtons();
+			}
+		}
+
+		private void UpdateButtons()
+		{
+			if (Element is Stepper stepper && Control is MaterialStepper control)
+			{
+				control.DecrementButton.Enabled = stepper.IsEnabled && stepper.Value > stepper.Minimum;
+				control.IncrementButton.Enabled = stepper.IsEnabled && stepper.Value < stepper.Maximum;
+			}
+		}
+
+		private void OnStep(object sender, EventArgs e)
+		{
+			if (Element is Stepper stepper && sender is MButton button)
+			{
+				var increment = stepper.Increment;
+				if (button == Control.DecrementButton)
+					increment = -increment;
+
+				stepper.SetValueFromRenderer(Stepper.ValueProperty, stepper.Value + increment);
+			}
+		}
+	}
+
+	public class MaterialStepper : UIView
```

Comment:
I am not sure where we want to put this - this is very this-renderer specific and has no use elsewhere. Basically - just for layout.